eng/emitters/pipelines/templates/stages/emitter-stages.yml

```diff
@@ -209,8 +235,19 @@ stages:
           - ${{ parameters.DependsOn }}
           - ${{ parameters.StagePrefix }}_Build
         variables:
-          selfRepositoryPath: $(Build.SourcesDirectory)
-          buildArtifactsPath: $(Pipeline.Workspace)/build_artifacts_${{ parameters.LanguageShortName }}
+          - ${{ each variable in parameters.Variables }}:
+              - ${{ if variable.key }}:
+                  # turn objects into name/value lists
+                  - name: ${{ variable.key }}
+                    value: ${{ variable.value }}
+              - ${{ else }}:
+                  # preserve all other variables
+                  - ${{ variable }}
+          - name: selfRepositoryPath
```

Comment:
I would add a blank line to separate the auto-splatted variables from the explicit variables